packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -173,19 +240,93 @@ class RawMenuAnchor extends StatefulWidget {
     this.consumeOutsideTaps = false,
     this.onOpen,
     this.onClose,
+    this.onOpenRequested = _defaultOnOpenRequested,
+    this.onCloseRequested = _defaultOnCloseRequested,
     this.useRootOverlay = false,
     this.builder,
     required this.controller,
     required this.overlayBuilder,
     this.child,
   });
 
-  /// A callback that is invoked when the menu is opened.
+  /// Called when the menu overlay is shown.
+  ///
+  /// This callback is triggered when the menu overlay is inserted into the
+  /// widget tree, typically before any opening animations begin. A typical
+  /// usage is to respond when the menu first becomes interactive, such as by
+  /// setting focus to a menu item.
+  ///
+  /// This callback is called regardless of whether the menu is already open or
+  /// the menu's position has changed.
```

Comment:
Despite the last paragraph, the first sentence seems a bit wrong: what do you mean by inserting the overlay to the widget tree while the overlay is already showing?

What do you think the rephrase like this?
```suggestion
  /// This callback is triggered every time [MenuController.open] is called -
  /// even if the overlay is already showing - and after any potential delay
  /// applied by [onOpenRequested] (that is, after its `showOverlay`). By the
  /// time [onOpen] is called, the overlay widget (the widget built by
  /// [overlayBuilder]) has been inserted to the widget tree.
  /// 
  /// A typical usage is to respond when the menu first becomes interactive,
  /// such as by setting focus to a menu item.
```

Reply:
The top paragraph is a little confusing if the reader has not read about onOpenRequested.

What about this:

```dart

  /// When [MenuController.open] is called, [onOpenRequested] is invoked with a
  /// `showOverlay` callback that, when called, shows the menu overlay and triggers
  /// [onOpen].
  ///
  /// The default implementation of [onOpenRequested] calls `showOverlay`
  /// synchronously, thereby calling [onOpen] synchronously. The [onOpen]
  /// callback is called regardless of whether the menu overlay was already
  /// showing or not.
  ///
  /// Custom implementations of [onOpenRequested] can delay the call to
  /// `showOverlay` or not call it at all, in which case [onOpen] will not be
  /// called.
```